internal/utils/utils_test.go

```diff
@@ -72,3 +82,217 @@ func TestGetCurrentNamespace(t *testing.T) {
 		})
 	}
 }
+
+func TestUpgradeCRDs(t *testing.T) {
+	t.Parallel()
+
+	manifests := []string{
+		`apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  name: hcpauths.secrets.hashicorp.com
+spec:
+  group: secrets.hashicorp.com
+  names:
+    kind: HCPAuth
+    listKind: HCPAuthList
+    plural: hcpauths
+    singular: hcpauth
+  scope: Namespaced
+  versions:
+  - name: v1beta1
+`,
+	}
+
+	manifestsUpdated := []string{
+		`apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  name: hcpauths.secrets.hashicorp.com
+spec:
+  group: secrets.hashicorp.com
+  names:
+    kind: HCPAuth
+    listKind: HCPAuthList
+    plural: hcpauths
+    singular: hcpauth
+  scope: Cluster
+  versions:
+  - name: v1beta2
+`,
+	}
+
+	others := []string{
+		`apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: foo
+`,
+	}
+
+	wantCRD := &apiextensionsv1.CustomResourceDefinition{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: apiextensionsv1.SchemeGroupVersion.String(),
+			Kind:       "CustomResourceDefinition",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "hcpauths.secrets.hashicorp.com",
+		},
+		Spec: apiextensionsv1.CustomResourceDefinitionSpec{
+			Group: "secrets.hashicorp.com",
+			Names: apiextensionsv1.CustomResourceDefinitionNames{
+				Kind:     "HCPAuth",
+				ListKind: "HCPAuthList",
+				Plural:   "hcpauths",
+				Singular: "hcpauth",
+			},
+			Scope: "Namespaced",
+			Versions: []apiextensionsv1.CustomResourceDefinitionVersion{
+				{
+					Name: "v1beta1",
+				},
+			},
+		},
+	}
+
+	wantCRDUpdated := wantCRD.DeepCopy()
+	wantCRDUpdated.Spec.Scope = "Cluster"
+	wantCRDUpdated.Spec.Versions = []apiextensionsv1.CustomResourceDefinitionVersion{
+		{
+			Name: "v1beta2",
+		},
+	}
+
+	scheme := runtime.NewScheme()
+	require.NoError(t, apiextensionsv1.AddToScheme(scheme))
+
+	root := t.TempDir()
+	ctx := context.Background()
+	builder := fake.NewClientBuilder().WithScheme(scheme)
+	tests := []struct {
+		name           string
+		c              client.Client
+		create         []*apiextensionsv1.CustomResourceDefinition
+		createDir      bool
+		manifests      []string
+		otherManifests []string
+		subDirs        int
+		canaries       int
```

Comment:
What are `canaries` in these tests? It looks like it's always set when `subDirs` is set in the test cases.